Automattic/wp-calypso
# Signup: passwordless onboarding

## Changes proposed in this Pull Request

We want to test the hypothesis that simplifying our account step for new users to only show an email address (and not require a password or username) will increase the number of people completing signup.

We will know this is true if we see an increase in the number of sites being created, and (hopefully) revenue increase.

<img width="300" alt="Screen Shot 2019-07-28 at 2 28 57 pm" src="https://user-images.githubusercontent.com/6458278/62002276-20eff500-b144-11e9-8e13-dde281f42a3a.png">

## TODO
- [x] Add 'account already created' notice if the step is complete
- [x] Add Google sign in button
- [x] Further testing with google account
- [x] Test with plan/domain purchase journeys
- [x] ~Check e2e tests (turn off this variation)~ No need any more :) p4TIVU-9eh-p2

## Testing instructions

ℹ️ For internal testing purposes, we'll launch this PR with the control set to 100% for a few days. Let's see if it survives :)

Engage `passwordless` variation for the `passwordlessSignup` AB test.

<img width="355" alt="Screen Shot 2019-07-29 at 4 46 59 pm" src="https://user-images.githubusercontent.com/6458278/62027229-831e2800-b220-11e9-9b17-25bd8456553f.png">

### Test cases

1. **Signup with a new email address and create a site**:
    a. You should be able to create an account and a site as per normal.
    b. You should be able to create an account, pay for a plan and domain as usual
    c. You should receive an activation email that explains about your generated username and (lack of) password^
    d. There should be an account activation task in the checklist
2. **Attempt signup with an exisiting account** - You be notified that you have an existing account and be prompted to log in. If the account is passwordless, you'll be sent a Magic Link.
3. **Attempt to sign in/sign up and create a site with a Google account** - You should be able to continue as per normal.
4. **On a flow other than onboarding** - Even with the AB variant set to `passwordless` you should never see the passwordless form

#### Relevant tracking events

`calypso_signup_actions_onboarding_passwordless_login_success` and `calypso_signup_actions_onboarding_passwordless_login_fail`- fired when the user submits the passwordless (`create-account`) step. Important Parameters:

```
action_message: "Successful login" | "An account with this email address already exists."
username: {The generated username (if successful)}
```
`calypso_signup_step_start` - fired when a step is loaded
`calypso_signup_actions_submit_step` - fired when all signup steps are submitted

Fixes https://github.com/Automattic/zelda-private/issues/107


^ Activation email

<img width="400" alt="Screen Shot 2019-09-24 at 9 14 02 am" src="https://user-images.githubusercontent.com/6458278/65485851-4a2fc600-dee6-11e9-9136-ce6e85605efa.png">
